driver/generic/st77xx.py

```diff
@@ -0,0 +1,404 @@
+            (ST77XX_INVCTR,b'\x00'),
+            (ST7735_PWCTR1,b'\x02\x70'),
+            (ST7735_PWCTR2,b'\x05'),
+            (ST7735_PWCTR3,b'\x01\x02'),
+            (ST7735_VMCTR1,b'\x3c\x38'),
+            (ST7735_PWCTR6,b'\b11\b15'),
+            # (ST77XX_GMCTRP1,b'\
+            ## memory access direction
+            # (ST77XX_MADCTL, bytes([ST77XX_MADCTL_ROTS[self.rot%4]]), 0),
+            # inverted on (?)
+            #(ST77XX_INVON, None, 10),
+            # normal display on
+            (ST77XX_NORON, None, 10),
+            # display on
+            (ST77XX_DISPON, None,100)
+        ]
+        if self.model in ('redtab','blacktab'): self._run_seq(init7735r)
+        else:
+            print('Warning: the greentab model was never properly tested')
+            self._run_seq(init7735)
+        # this applies ST77XX_MADCTL
+        self.apply_rotation(self.rot)
+
+
+class St7789(St77xx):
+    def __init__(self,res,**kw):
+        super().__init__(res=res,suppRes=[(240,320),],model=None,suppModel=None,**kw)
+    def config(self):
+        init7789=[
+            # out of sleep mode
+            (ST77XX_SLPOUT, None, 100),
+            # memory access direction (this is set again in apply_rotation, is that okay?)
+            (ST77XX_MADCTL, bytes([ST77XX_MADCTL_BGR | ST77XX_MADCTL_ROTS[self.rot%4]])),
+            # RGB565
+            (ST77XX_COLMOD, bytes([ST77XX_COLOR_MODE_65K | ST77XX_COLOR_MODE_16BIT])),
+            # front/back porch setting in normal/idle/partial modes; 3rd byte (PSEN) 0x00 = disabled
+            (ST7789_PORCTRL, b"\x0C\x0C\x00\x33\x33"),
+            # VGH=14.06V, VGL=-8.87V [Adafruit: 0x14]
+            (ST7789_GCTRL, b"\x35"),
+            # [Adafruit: missing]
+            (ST77XX_DISPOFF, b"\x28"),
+            # power control [Adafruit: 0x2c]
+            (ST7789_LCMCTRL, b"\x3C"),
+            # power control (set VDV and VRD by register write), write VRH and VDV
+            (ST7789_VDVVRHEN, b"\x01"),(ST7789_VRHS, b"\x0B"),(ST7789_VDVS, b"\x20"),
+            # frame rate 60Hz
+            (ST7789_FRCTRL2, b"\x0F"),
+            # power control: AVDD=6.6V, AVCL=-4.8V, VDS=2.4V
+            (ST7789_PWCTRL1, b"\xA4\xA1"),
+            # positive voltage gamma control
+            (ST7789_PVGAMCTRL, b"\xD0\x01\x08\x0F\x11\x2A\x36\x55\x44\x3A\x0B\x06\x11\x20"),
+            # negative voltage gamma control
+            (ST7789_NVGAMCTRL, b"\xD0\x02\x07\x0A\x0B\x18\x34\x43\x4A\x2B\x1B\x1C\x22\x1F"),
+            # content adaptive brightness control and color enhancement: color enhancement on, high enhancement
+            (ST7789_WRCACE, bytes([0b1011_0000])),
+            # display on
+            (ST77XX_DISPON, None, 100),
+            ## FIXME: needs out of sleep mode AGAIN, otherwise will stay bleck the first time on?
+            (ST77XX_SLPOUT, None, 100),
+        ]
+        self._run_seq(init7789)
+        self.apply_rotation(self.rot)
+
+
+if __name__=='__main__':
```

Comment:
I don't think test code belongs here.   
You can add it as a separate file under examples.  
I also think it would make more sense to provide some LVGL example instead of these rectangles.

Reply:
I can put it aside, yes.

The hardware driver is not bound to LVGL (and I think it should not be; it is indeed the problem of most python drivers for those displays that they work only for a specific framework, for not much reason TBH), and the rectangles are meant to verify that screen edges have correct offsets — this depends on the display model, and would be much more difficult to see with LVGL widgets.

This is also relevant to whether the driver should be registered with LVGL in the ctor. I'd say no, or perhaps in a derived class which would be a thin wrapper around the hardware driver itself. (And again, this wrapper could be generic and provided by LVGL, but IIRC generic driver interface was already discussed somewhere, so it is just a digression here).

Reply:
If you want a driver that is not bound to LVGL, please also provide a wrapper that is LVGL specific.   
The derived LVGL wrapper should register LVGL display driver, handle event loop etc.  This is not something we expect LVGL users to do.